lua/plenary/path.lua

```diff
@@ -229,6 +240,51 @@ function Path:rmdir()
   uv.fs_rmdir(self:absolute())
 end
 
+function Path:touch(opts)
+  opts = opts or {}
+
+  local mode = opts.mode or 420
+  local parents = F.if_nil(opts.parents, false, opts.parents)
+
+  if self:exists() then return end
+
+  if parents then
+    -- alt: split with self._sep take everything besides last entry. @tj?
+    local only_dirs = self:absolute():match(string.format('^(.+)%s([^%s]+)', self._sep, self._sep))
+    Path:new(only_dirs):mkdir({ parents = true })
+  end
+
+  local fd = vim.loop.fs_open(self:absolute() or self.filename, "w", mode)
```

Comment:
`self:absolute() or self.filename` looks like a pattern. Maybe `self:_fs_filename()` or something like that.